qutebrowser/browser/history.py

```diff
@@ -71,162 +71,81 @@ def url_str(self):
         """Get the URL as a lossless string."""
         return self.url.toString(QUrl.FullyEncoded | QUrl.RemovePassword)
 
-    @classmethod
-    def from_str(cls, line):
-        """Parse a history line like '12345 http://example.com title'."""
-        data = line.split(maxsplit=2)
-        if len(data) == 2:
-            atime, url = data
-            title = ""
-        elif len(data) == 3:
-            atime, url, title = data
-        else:
-            raise ValueError("2 or 3 fields expected")
 
-        url = QUrl(url)
-        if not url.isValid():
-            raise ValueError("Invalid URL: {}".format(url.errorString()))
+class CompletionHistory(sql.SqlTable):
 
-        # https://github.com/qutebrowser/qutebrowser/issues/670
-        atime = atime.lstrip('\0')
+    """History which only has the newest entry for each URL."""
 
-        if '-' in atime:
-            atime, flags = atime.split('-')
-        else:
-            flags = ''
-
-        if not set(flags).issubset('r'):
-            raise ValueError("Invalid flags {!r}".format(flags))
-
-        redirect = 'r' in flags
+    def __init__(self, parent=None):
+        super().__init__("CompletionHistory", ['url', 'title', 'last_atime'],
```

Comment:
Oh, I just noticed this causes the completion to show "CompletionHistory" as title... I guess we either need a way to set the title separately, or we need to rename this to `History` and the other one to `FullHistory` or so...